redis_cache/__init__.py

```diff
@@ -158,6 +168,37 @@ def invalidate(self, *args, **kwargs):
         pipe.zrem(self.keys_key, key)
         pipe.execute()
 
+    def invalidate_partial(self, *args, **kwargs):
+        """Invalidate items in the cache based on partial matches.
+
+        A partial match is when any of the values in ``args`` are in the key's
+        ``args`` or all of the key/value pairs in ``kwargs`` are in the key's
```

Comment:
This is a little strange. Why not all args and kwargs?

Reply:
Imagine I have a function that does something like this:

```python
def calc_total_weight(*objs: list[dict]):
    n = 0
    for obj in args:
        n += obj["weight"]
    return n
```

If one of the `objs` has changed, I want to invalidate any response from `calc_total_weight` that contained that one.

Reply:
Another use case is

```python
def get_object_weight(obj_id, as_percent=False):
    pass
```

In this case, if the object in question changed, I would use `invalidate_partial(obj_id=x)` to invalidate cached responses for both values of `as_percent`.

Reply:
Right, but what about the issue with something like this?
```python
def do_something(user_id, tenant_id):
  pass
```

How do you remove the cache for this one? Assuming they are called as `args`.

Reply:
I don't believe this type of function is well supported because the ids have to be passed in the same order
> def calc_total_weight(*obj_ids: list[dict]):

Reply:
> Right, but what about the issue with something like this?
> 
> ```python
> def do_something(user_id, tenant_id):
>   pass
> ```
> 
> How do you remove the cache for this one? Assuming they are called as `args`.

You're right. Calling `invalidate_partial(5)` would invalidate the cache for `user_id=5` and `tenant_id=5`, so should not be used unless the author really wanted that kind of behavior.

Reply:
> I don't believe this type of function is well supported because the ids have to be passed in the same order
> 
> > def calc_total_weight(*obj_ids: list[dict]):

I just added a test for this case and got it working properly for it.

Reply:
> Right, but what about the issue with something like this?
> 
> ```python
> def do_something(user_id, tenant_id):
>   pass
> ```
> 
> How do you remove the cache for this one? Assuming they are called as `args`.

Actually it seems like this case is now covered in my latest commit. One would simply call `do_something.invalidate_partial(user_id=5)` to invalidate all cache entries for that user regardless of `tenant_id`.